cpp/include/cudf/replace.hpp

```diff
@@ -16,11 +16,70 @@
 
 #pragma once
 
-#include "cudf/cudf.h"
-#include "cudf/types.hpp"
-#include "cudf/column/column.hpp"
+#include <cudf/column/column.hpp>
+#include <cudf/scalar/scalar.hpp>
+#include <rmm/mr/device_memory_resource.hpp>
+#include <rmm/mr/default_memory_resource.hpp>
+
+// Forward declaration
+typedef struct CUstream_st* cudaStream_t;
 
 namespace cudf {
+namespace experimental {
+/**
+  * @brief Replaces all null values in a column with corresponding values of another column.
+  *
+  * Returns a column `output` such that if `input[i]` is valid, its value will be copied to
+  * `output[i]`. Otherwise, `replacements[i]` will be copied to `output[i]`.
+  *
+  * The `input` and `replacement` columns must be of same size and have the same
+  * data type.
+  *
+  * @param[in] input A cudf::column containing null values
+  * @param[in] replacement A cudf::column whose values will replace null values in input
+  *
+  * @returns Column with nulls replaced
+  */
+std::unique_ptr<cudf::column> replace_nulls(cudf::column_view const& input,
+                                            cudf::column_view const& replacement,
+                                            rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+
+/**
+  * @brief Replaces all null values in a column with a scalar.
+  *
+  * Returns a column `output` such that if `input[i]` is valid, its value will be copied to
+  * `output[i]`. Otherise, `replacement` will be coped to `output[i]`.
+  *
+  * `replacement` must have the same data type as `input`.
+  *
+  * @param[in] input A cudf::column containing null values
+  * @param[in] replacement A cudf::scalar whose value will replace null values in input
+  *
+  * @returns Column with nulls replaced
+  */
+std::unique_ptr<cudf::column> replace_nulls(cudf::column_view const& input,
+                                            cudf::scalar const* replacement,
+                                            rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
+
+
+/**
+ * @brief Replace elements from `input_col` according to the mapping `old_values` to
```

Comment:
Bit too long for a brief. 
```suggestion
 * @brief Finds and replaces values within a column.
 * 
 * Creates a copy of @p input, but replace elements according to the mapping `old_values` to
```